src/Aspire.Hosting/ApplicationModel/CustomResourceSnapshot.cs

```diff
@@ -18,6 +18,8 @@ public sealed record CustomResourceSnapshot
     private readonly ImmutableArray<HealthReportSnapshot> _healthReports = [];
     private readonly ResourceStateSnapshot? _state;
 
+    internal long Id { get; set; }
```

Comment:
init instead of set?
```suggestion
    // An internal id for debugging purposes
    internal long Id { get; init; }
```